akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -267,7 +271,10 @@ private[stream] final class GraphInterpreter(
   def finish(): Unit = {
     var i = 0
     while (i < logics.length) {
-      if (!isStageCompleted(i)) logics(i).postStop()
+      if (!isStageCompleted(i)) {
+        logics(i).postStop()
```

Comment:
I will do these in another PR, these are the missing tests/features on the interpreter I mentioned earlier. I am still working with a half-tested, half-done engine :D (living dangerous)